runtime/evaluate.cc

```diff
@@ -195,6 +195,18 @@ class evaluator : public expr_visitor, public stmt_visitor {
     }
   }
 
+  index_t eval_short_circuit_op(const call* op) {
```

Comment:
Shouldn't this return `bool`?

Reply:
I had been thinking that all these evaluators return `index_t` and the function is responsible for promoting. But now I see I did the opposite for `buffer_at` which returns a pointer. Changed it to return `bool`.